packages/react/ds/src/cookie-banner/cookie-banner.tsx

```diff
@@ -0,0 +1,87 @@
+'use client';
+import React, { useState, cloneElement } from 'react';
+import { ButtonProps } from '../button/types.js';
+import { Container } from '../container/container.js';
+import { LinkProps } from '../link/link.js';
+
+type CookieBannerProps = {
+  children: React.ReactNode;
+  accept: {
+    children: React.ReactNode;
+    triggerButton: React.ReactElement<ButtonProps>;
+  };
+  reject: {
+    children: React.ReactNode;
+    triggerButton: React.ReactElement<ButtonProps>;
+  };
+  dismissButton?: React.ReactElement<ButtonProps>;
+  cookieLink?: React.ReactElement<LinkProps>;
+};
+
+export const CookieBanner = ({
+  children,
+  accept,
+  reject,
+  cookieLink,
+  dismissButton,
+}: CookieBannerProps) => {
+  const [status, setStatus] = useState<null | boolean | 'dismiss'>(null);
+  const handleOpenBanner = () => setStatus(true);
+  const handleCloseBanner = () => setStatus(false);
+  const handleDismissBanner = () => setStatus('dismiss');
+  return (
+    <>
+      {status === 'dismiss' ? null : (
+        <div className="gi-cookie-banner-container">
+          <div className="gi-py-5">
+            <Container>
+              {/* default container of cookie banner   */}
+              {status === null && (
+                <>
+                  {children}
+                  <div className="gi-cookie-banner-buttons">
+                    {cloneElement(accept.triggerButton, {
+                      onClick: handleOpenBanner,
+                    })}
+                    {cloneElement(reject.triggerButton, {
+                      onClick: handleCloseBanner,
+                    })}
+                    {cookieLink && cookieLink}
+                  </div>
+                </>
+              )}
+
+              {/* accepted container of cookie banner */}
+              {status === true && (
```

Comment:
suggestion: just status would be a lot cleaner?

Reply:
the status has the following type `<null | boolean | 'dismiss'>` if I set it to just `status` it will check if it's truthy, which will be true for `true` and `dismiss`